pkgs/tools/archivers/7zz/default.nix

```diff
@@ -56,8 +56,45 @@ stdenv.mkDerivation rec {
   patches = [ ./fix-build-on-darwin.patch ];
   patchFlags = [ "-p0" ];
 
+  postPatch = lib.optionalString stdenv.hostPlatform.isMinGW ''
+    find . \( -iname '*.cpp' -o -iname '*.h' -o -iname '*.c' -o -iname '*.rc' \) -print0 \
```

Comment:
This is way too brittle. We usually try to apply arch-specific patches unconditionally (see how `fix-build-on-darwin.patch` is applied unconditionally), but this here looks like it's a problem with the mingw environment and path case sensitivity in general, which might be better fixed more globally?

Reply:
Thanks for the review and suggestion. I've converted this script to a patch, going for the less generic solution because I haven't been able to find any other packages with similar workarounds or issues.

Reply:
I checked the patch. Indeed all paths seem to be in the windows-only branches.

Could it be that upstream simply got the case wrong, and because they were mostly not cross-compiling, but building on a case-insensitive filesytem it didn't surface so far?

Reply:
Can you send the patch to https://sourceforge.net/p/sevenzip/patches/, linking to here for context?

Reply:
The case of these headers and libraries isn't wrong, it's just meant for a case-insensitive filesystem. Microsoft's documentation itself uses inconsistent casing. The ability to use camel case to make header names more readable could be considered a feature!

I've sent the patch upstream https://sourceforge.net/p/sevenzip/patches/406/

Reply:
Thanks!